### PR TITLE
chore: add discussions link

### PR DESCRIPTION
**Pull request by henryiii**
This sets up Discussions, and uses the Issue forms tooling from GitHub. See a live preview here: https://github.com/henryiii/pybind11/issues/new/choose

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -176,3 +176,5 @@ to the terms and conditions of this license.
    :target: https://repology.org/project/python:pybind11/versions
 .. |Python Versions| image:: https://img.shields.io/pypi/pyversions/pybind11.svg
    :target: https://pypi.org/project/pybind11/
+.. |GitHub Discussions| image:: https://img.shields.io/static/v1?label=Discussions&message=Ask&color=blue&logo=github
+   :target: https://github.com/scikit-hep/boost-histogram/discussions
```
Comment (rwgk) [inline README.rst:180]: Is this correct? (link to skikit-hep)

Comment (henryiii) [reply README.rst:180]: Nope, copy-paste typo!

**Review by rwgk — APPROVED** [review]
Awesome, thanks for driving this!

**Comment by henryiii** [comment]
Okay, let's try it out!

**Pull request merged**